src/material-experimental/mdc-list/list-item.html

```diff
@@ -1,16 +1,33 @@
-<ng-content select="[mat-list-avatar],[matListAvatar],[mat-list-icon],[matListIcon]"></ng-content>
+<!-- Divide content into templates -->
 
-<!-- If lines were explicitly given, use those as the text. -->
-<ng-container *ngIf="lines.length">
-  <span class="mdc-list-item__text"><ng-content select="[mat-line],[matLine]"></ng-content></span>
-</ng-container>
+<ng-template #prefix>
+  <ng-content select="[mat-list-avatar],[matListAvatar],[mat-list-icon],[matListIcon]"></ng-content>
+</ng-template>
```

Comment:
The pattern of `ng-content` inside `ng-template` is problematic because the content gets "created" even when this `ng-template` isn't stamped out anywhere, so it incurs extra performance/memory costs (the content gets created _again_ when it's actually stamped out). This includes putting an `ng-content` inside of an `ngIf` / `ngFor`. If it's possible to get the same result just using `ng-content` by itself, that would be preferable.

Reply:
The templates do all get stamped out in the end, so that part of it at least isn't a concern. For the list-item template, doing it with just conditional class bindings was already getting difficult to understand. For the list-option, its probably impossible to do it without putting some of the content in `*ngIf`/`ng-template`. We could reduce the degree to which we use it, but the code would be pretty hard to understand/maintain. Unless we've ran into actual performance issues due to this in the past, it just seems like premature optimization at the cost of maintainability to me.

Reply:
We certainly received complaints about it in the past, particularly with tabs (where we do this), though granted the content of a list item will be typically less complex than a tab. Thinking about it more, the performance stuff is bad, but the worse thing is that the behavior is really _surprising_ for users of the component because any directives/components they put in the content will be instantiated extra times. This can make debugging more confusing if someone gets the instances mixed up.

Reply:
Ok I'll see what I can do. I don't think I can completely eliminate `ngIf` and `ng-template`, but I'm sure I can reduce the number